src/main/java/org/apache/maven/plugins/enforcer/EnforceBytecodeVersion.java

```diff
@@ -331,6 +331,10 @@ private String isBadArtifact( Artifact a )
         {
             throw new EnforcerRuleException( "IOException while reading " + jarFile.getName(), e );
         }
+        catch ( IllegalArgumentException e )
+        {
+            throw new EnforcerRuleException( "Error while reading " + jarFile.getName(), e );

```

Comment:
Following the same logic as in #27, probably this should use `f.getAbsolutePath()` instead of `jarFile.getName()` in case the jarFile couldn't be opened. (Or both and carefully protecting the `jarFile` call).

Reply:
Well at this point that makes no sense, because the `IllegalArgumentException` is thrown only under conditions when the `JarFile` _has_ been opened successfully, Anyway, the suggested change is fine.